### ISSUE TITLE
Pluto typechecker (opinion)

### ISSUE THREAD

**Issue by cstml** [Plutonomicon/pluto#4]
Hi,

Thank you for creating/working on/sharing  `Pluto`.  

What are your thoughts on integrating a `System F` type-checker with the current Pluto implementation?(of course, with type erasure) Do you consider it a feasible/good idea?

**Comment by L-as** [comment]
I've discussed this with Morgan, and System F isn't a good idea because we need type inference too.

**Comment by morganthomas** [comment]
Right, but to elaborate further, I do want a type checker, I think it's a good idea, but it's kind of a lot of work. It's not a lot of work to implement a basic type system. What is a lot of work is if you expect it to have all the conveniences that we usually expect, like type inference and pattern matching. This project is about two weeks old at this point, and we needed to get something up and running fast, so that's why there are no types.
